### ISSUE TITLE
Loss not changing

### ISSUE THREAD

**Issue by Devarsh-UTD** [hyperopt/hyperopt#497]
Hello,
I do not know if this is the right way to use Hyperopt, but based on my readings below is how i am using it, if you think it is incorrect kindly let me know .
```
def _get_best_param(self, cross_val_temp_feats_train_valid, xgb_feats):
        logging.info('Orders: getting best params:')
        def score(space):
            print("Training with params: ")
            print(space)
            group = cross_val_temp_feats_train_valid.date_combined
            gkf = GroupKFold(n_splits=10).split(cross_val_temp_feats_train_valid[xgb_feats],
                                                cross_val_temp_feats_train_valid['source'], group)
            estimator = XGBRegressor(random_state=1234, n_jobs=-1, params=space)
            score = cross_val_score(estimator, cross_val_temp_feats_train_valid[xgb_feats],
                                    cross_val_temp_feats_train_valid['source'],
                                    cv=gkf, n_jobs=-1, scoring='neg_mean_squared_error')
            cv_score = np.mean(np.abs(score))
            print("SCORE:", cv_score)
            return {'loss': cv_score, 'status': STATUS_OK}

        space = {
            'n_estimators': hp.quniform('n_estimators', 100, 1000, 1),
            'learning_rates': hp.quniform('eta', 0.025, 0.5, 0.025),
            'max_depth': hp.choice('max_depth', np.arange(3, 15, dtype=int)),
            'min_child_weight': hp.quniform('min_child_weight', 1, 7, 1),
            'subsample': hp.quniform('subsample', 0.5, 1, 0.05),
            'gamma': hp.quniform('gamma', 0.5, 1, 0.05),
            'colsample_bytree': hp.quniform('colsample_bytree', 0.5, 1, 0.05),
            'eval_metric': 'rmse',
            'objective': 'reg:linear',
            'booster': 'gbtree',
            'silent': 1,
            'seed': 1234
        }
        # Use the fmin function from Hyperopt to find the best hyperparameters
        seed = 1234
        best = fmin(score, space, algo=tpe.suggest,
                    max_evals=50, rstate= np.random.RandomState(seed))
        print("The best hyperparameters are: ", "\n")
        print(best)
        return best
```

In the above code, 
I am doing cross validation for a particular set of hyperparameters, and returning the mean score for that set of hyperparameter. 
My problem is - The loss computed(MSE) for a particular set of hyperparameters, does not change at all , even when differnet set of hyperparameter is used. Am I doing something wrong?, kinldy help me out. Thnx

**Comment by Devarsh-UTD** [comment]
Well, I corrected it.
In the above code, I am passing parameters to XGBoost using `params=space`, because of which certain parameters in the estimator take their default value, rather than the values being supplied from outside. Hence the loss was remaining constant over different iterations.

**Comment by leon-dae** [comment]
Just in case somebody else finds this. 
I had the same problem, but in my case I initiated the trials=Trials() _before_ starting the cross-validation. However that should be done _within every iteration_ of the cross-validation.

**Comment by rietesh** [comment]
https://stackoverflow.com/questions/69520619/xgboost-loss-not-changing-with-hyperopt This remains unsolved any ideas?
